botnim/vector_store/search_modes.py

```diff
@@ -0,0 +1,183 @@
+            boost_factor=1.0,
+            field_path="metadata.extracted_data.AdditionalKeywords"
+        ),
+        SearchFieldConfig(
+            name="topics",
+            weight=FieldWeight(exact_match=0.4, partial_match=0.4, semantic_match=1.0),
+            boost_factor=1.0,
+            field_path="metadata.extracted_data.Topics"
+        ),
+    ]
+)
+
+# Canonical default search mode for all business logic
+DEFAULT_SEARCH_MODE_NAME = "REGULAR"
+DEFAULT_SEARCH_MODE = REGULAR_CONFIG
+
+RELATED_RESOURCE_CONFIG = SearchModeConfig(
+    name="RELATED_RESOURCE",
+    description="Finds documents that are related to or mention a specific resource, by matching ReferenceLinks, LegalReferences, and related fields. Uses exact phrase matching for reference fields.",
+    min_score=0.5,
+    num_results=5,
+    use_vector_search=False,
+    fields=[
+        SearchFieldConfig(
+            name="reference_links",
+            weight=FieldWeight(exact_match=15.0, partial_match=0.0, semantic_match=0.0),
+            boost_factor=10.0,
+            field_path="metadata.extracted_data.ReferenceLinks",
+            use_phrase_match=True
+        ),
+        SearchFieldConfig(
+            name="legal_references_title",
+            weight=FieldWeight(exact_match=12.0, partial_match=0.0, semantic_match=0.0),
+            boost_factor=8.0,
+            field_path="metadata.extracted_data.LegalReferences.ReferenceTitle",
+            use_phrase_match=True
+        ),
+        SearchFieldConfig(
+            name="legal_references_text",
+            weight=FieldWeight(exact_match=10.0, partial_match=0.0, semantic_match=0.0),
+            boost_factor=6.0,
+            field_path="metadata.extracted_data.LegalReferences.ReferenceText",
+            use_phrase_match=True
+        ),
+        SearchFieldConfig(
+            name="description",
+            weight=FieldWeight(exact_match=2.0, partial_match=1.0, semantic_match=1.0),
+            boost_factor=1.0,
+            field_path="metadata.extracted_data.Description"
+        ),
+    ]
+)
+
+# Immutable registry of all search modes
+SEARCH_MODES = MappingProxyType({
+    "TAKANON_SECTION_NUMBER": TAKANON_SECTION_NUMBER_CONFIG,
+    "REGULAR": REGULAR_CONFIG,
+    "RELATED_RESOURCE": RELATED_RESOURCE_CONFIG,
+    # Add more modes here as needed
+})
+
+def normalize_resource_identifier(resource: str) -> str:
+    """
+    Normalize a resource identifier (e.g., law name + section) for consistent matching.
+    - Strips whitespace
+    - Converts common punctuation to spaces
+    - Normalizes section formatting (e.g., 'סעיף 9א' -> 'סעיף 9א')
+    - Lowercases (if appropriate)
+    - Removes redundant characters
+    """
+    import re
+    s = resource.strip()
+    s = re.sub(r'[\-_,:]', ' ', s)  # Replace common punctuation with space
+    s = re.sub(r'\s+', ' ', s)     # Collapse multiple spaces
+    s = s.replace(' סעיף ', ' סעיף')  # Remove space before 'סעיף'
```

Comment:
why are we doing this and why isn't strip() enough?

Reply:
not sure I understand the question:. We do it to collapse multiple spaces, replace punctuation, or handle quirks

Reply:
you are right of course :( 

removed the function, and will never commit again without at least 5h of sleep.